FileDescriptorMeter.c

```diff
@@ -33,11 +35,42 @@ static void FileDescriptorMeter_updateValues(Meter* this) {
    /* only print bar for first value */
    this->curItems = 1;
 
-   /* Use maximum value for scaling of bar mode */
-   this->total = this->values[1];
+   /* Use maximum value for scaling of bar mode
+    *
+    * As the plain total value can be very large compared to
+    * the actually used value, this is capped in the following way:
+    *
+    * 1. If the maximum value is below (or equal to) 1<<16, use it directly
+    * 2. If the maximum value is above, use powers of 2 starting at 1<<16 and
+    *    double it until it's larger than 16 times the used file handles
+    *    (capped at the maximum number of files)
+    * 3. If the maximum is effectively unlimited (AKA > 1<<30),
+    *    Do the same as for 2, but cap at 1<<30.
+    */
+   if (this->values[1] <= 1<<16) {
+      this->total = this->values[1];
+   } else {
+      if (this->total < 16 * this->values[0]) {
+         for (this->total = 1<<16; this->total < 16 * this->values[0]; this->total *= 2) {
+            if (this->total >= 1<<30) {
+               break;
+            }
+         }
```

Comment:
I can't say if this optimization is worth it, but here it is:
```suggestion
         int exp;
         (void) frexp(16 * this->values[0], &exp);
         this->total = ldexp(1.0, MINIMUM(exp, 16));
```

Reply:
Even if this was faster I find it kinda hard to read/understand.